File: Data Structures & Algorithms/same-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q == null) {
            System.out.println("loop 1");
            return true;
        }

        if(p == null || q == null) {
            System.out.println("loop 2");
            return false;
        }
        if(p.val != q.val) {
            System.out.println("loop 3");
            return false;
        }
           
        
        boolean val1 = isSameTree(p.left, q.left);
        boolean val2 = isSameTree(p.right, q.right);

        if(val1 == true && val2 == true) {
            return true;
        } 
        return false;

    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:

        return self.dfs(root, root.val)



    def dfs(self, root, max_so_far):
        if not root:
            return 0

        res = 0
        if root.val >= max_so_far:
            res = 1
            max_so_far = max(root.val, max_so_far)
        
        res += self.dfs(root.left, max_so_far)
        res += self.dfs(root.right, max_so_far)

        return res
        